File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/user/user/AppUserResponseDTO.java
package com.example.fieldsync_inventory_api.dto.user.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AppUserResponseDTO {
    private Integer id;
    private String username;
    private String password;
    private Boolean isActive;
    private Instant createdAt;
    private Instant updatedAt;
    private Instant deletedAt;
    private List<String> roles;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/app_user/AppUserResponseDto.kt
package com.example.fieldsync_inventory_app.data.remote.dto.app_user

data class AppUserResponseDto(
    val id: Int,
    val username: String,
    val password: String,
    val isActive: Boolean,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?,
    val roles: List<String>
)
